agent/api/container.go

```diff
@@ -210,3 +223,49 @@ func (c *Container) String() string {
 	}
 	return ret
 }
+
+// GetSteadyStateStatus returns the steady state status for the container. If
+// Container.steadyState is not initialized, the default steady state status
+// defined by `defaultContainerSteadyStateStatus` is returned. The 'pause'
+// container's steady state differs from that of other containers, as the
+// 'pause' container can reach its teady state once networking resources
+// have been provisioned for it, which is done in the `ContainerResourcesProvisioned`
+// state
+func (c *Container) GetSteadyStateStatus() ContainerStatus {
+	if c.steadyState == nil {
+		return defaultContainerSteadyStateStatus
+	}
+	return *c.steadyState
+}
+
+// IsKnownSteadyState returns true if the `KnownState` of the container equals
+// the `steadyState` defined for the container
+func (c *Container) IsKnownSteadyState() bool {
+	knownStatus := c.GetKnownStatus()
+	return knownStatus == c.GetSteadyStateStatus()
```

Comment:
Nope, GetSteadyStateStatus() has a pointer receiver type and combining these into a single line results in a compilation error because of that.